### ISSUE TITLE
Error when using within React Demo

### ISSUE THREAD

**Issue by guangyingmi** [replicate/replicate-javascript#40]
```javascript
import Replicate from "replicate";

const REPLICATE_API_TOKEN = "**************";

async function App() {
  const replicate = new Replicate({
    auth: REPLICATE_API_TOKEN,
  });
  const output = await replicate.run(
    "tstramer/midjourney-diffusion:436b051ebd8f68d23e83d22de5e198e0995357afef113768c20f0b6fcef23c8b",
    {
      input: {
        prompt: "a photo of an astronaut riding a horse on mars",
      },
    }
  );
  console.log(output);
  return (
    <div className="App">

    </div>
  );
}

export default App;

```

![image](https://user-images.githubusercontent.com/51470807/230575220-ac6a4200-2fb7-46e9-ad54-160253b72eb6.png)

**Comment by mattt** [comment]
Hi @guangyingmi. `async` functions can't be used as React components. Instead, you should create a class or function component and use state and side effects.

Here's an example of how to do that with `useState` and `useEffect` hooks:

```javascript
import React, { useState, useEffect } from 'react';
import Replicate from 'replicate';

const REPLICATE_API_TOKEN = '**************';

function App() {
  const [output, setOutput] = useState(null);

  useEffect(() => {
    (async () => {
      const replicate = new Replicate({
        auth: REPLICATE_API_TOKEN,
      });
      const result = await replicate.run(
        'tstramer/midjourney-diffusion:436b051ebd8f68d23e83d22de5e198e0995357afef113768c20f0b6fcef23c8b',
        {
          input: {
            prompt: 'a photo of an astronaut riding a horse on mars',
          },
        }
      );
      setOutput(result);
    })();
  }, []);

  return (
    <div className="App">
      {output ? (
        <div>
          <h2>Output:</h2>
          <pre>{JSON.stringify(output, null, 2)}</pre>
        </div>
      ) : (
        <p>Loading...</p>
      )}
    </div>
  );
}

export default App;
```

**Comment by mattt** [comment]
The error "TypeError: Failed to execute 'fetch' on 'Window': Illegal invocation" seems to indicate that you're using the Replicate Node.js library in a client-side React application, which isn't currently supported.

Please take a look at our ["Build a website with Next.js"](https://replicate.com/docs/get-started/nextjs) guide for information on how to get started.

**Comment by guangyingmi** [comment]
First of all, my intention was to use ChatGPT and midjourney to create a small web application that can interpret people's dreams and draw them out. However, I found that midjourney cannot directly call APIs, so I started looking for help on GitHub. Later, I stumbled upon your repository, but I have not been able to call it successfully, so I came here to raise an issue. Thank you very much for your response, even though I have not succeeded yet!
